src/Workspaces/Remote/ServiceHub/Services/LanguageServer.cs

```diff
@@ -94,29 +94,28 @@ public Task<SymbolInformation[]> WorkspaceSymbolAsync(WorkspaceSymbolParams args
 
         private async Task<ImmutableArray<SymbolInformation>> SearchAsync(Solution solution, WorkspaceSymbolParams args, CancellationToken cancellationToken)
         {
-            // When progress reporting is supported, report incrementally per project and return an empty result at the end.
-            // Otherwise aggregate and return the results for all projects at the end.
+            // When progress reporting is supported, report incrementally per document and return an empty result at the end.
+            // Otherwise aggregate and return the results for all documents at the end.
             if (args.Progress != null)
             {
-                var tasks = solution.Projects.Select(p => SearchProjectAndReportSymbolsAsync(p, args, cancellationToken));
+                var tasks = solution.Projects.SelectMany(p => p.Documents).Select(d => SearchDocumentAndReportSymbolsAsync(d, args, cancellationToken));
                 await Task.WhenAll(tasks).ConfigureAwait(false);
                 return ImmutableArray<SymbolInformation>.Empty;
             }
             else
             {
-                var tasks = solution.Projects.Select(p => SearchProjectAsync(p, args.Query, cancellationToken));
+                var tasks = solution.Projects.SelectMany(p => p.Documents).Select(d => SearchDocumentAsync(d, args.Query, cancellationToken));
                 var results = await Task.WhenAll(tasks).ConfigureAwait(false);
                 return results.SelectMany(a => a).ToImmutableArray();
```

Comment:
we're parallelizing the search.  but we still return all the values in one large batch.  That's still a problem given that it's the serialization of the result that is the expensive non-cancellable bit.  Can we not return the results in smaller pieces?

Reply:
should we just not support this codepath and say that we only support the 'progress' based one?

Reply:
Depends on how much we care about back compatibility.  For this feature in particular it's probably fine since it's just providing results in local VS (pending migration of C#_LSP -> C#), but in cloud env and liveshare it's possible to have clients on any number of versions, some of which wouldn't support it.

Reply:
> But in cloud env and liveshare it's possible to have clients on any number of versions, some of which wouldn't support it.

I'm ok ditching support for older clients.  It's just an additional thing to test and maintain.  This is all alpha code anyways, right?